src/events/log.js

```diff
@@ -11,29 +19,8 @@ exports.run = (client, data, type = "log") => {
   if (typeof data === "object" && typeof data !== "string" && !Array.isArray(data)) data = require("util").inspect(data, { depth: 0, colors: true });
   if (Array.isArray(data)) data = data.join["\n"];
 
-  let timestamp = "";
-  if (!client.config.disableLogTimestamps) {
-    timestamp = `[${moment().format("YYYY-MM-DD HH:mm:ss")}]`;
-    if (!client.config.disableLogColor) {
-      switch (type) {
-        case "debug":
-          timestamp = clk.bgMagenta(timestamp);
-          break;
-        case "warn":
-          timestamp = clk.black.bgYellow(timestamp);
-          break;
-        case "error":
-          timestamp = clk.bgRed(timestamp);
-          break;
-        case "log":
-          timestamp = clk.bgBlue(timestamp);
-          break;
-          // no default
-      }
-    }
-    timestamp += " ";
-  }
-
-  if (type === "debug") type = "log";
-  console[type](data.split("\n").map(str => timestamp + str).join("\n"));
+  const format = formats[type || "log"];
+  let time = client.config.disableLogTimestamps ? "" : `[${moment().format("YYYY-MM-DD HH:mm:ss")}]`;
+  if ("time" in format) time = format.type(time);
```

Comment:
`format.type` will be undefined here.

Reply:
There's also no reason to check if time is inside format since all 6 objects have the time property in them.

Reply:
yep that was a typo :/
the check is there just in case someone in the future will add a log with no timestamp coloring.

Fixed both in the new commit